### PR TITLE
Prepare changelog for Release v0.35.8

### PR DESCRIPTION
**Pull request by creachadair**

### CODE REVIEW

**Review by tychoish — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,35 @@
...
+
+July 12, 2022
+
+Special thanks to external contributors on this release: @joeabbey
+
+### BREAKING CHANGES
+
+- CLI/RPC/Config
+
+   - [node] [\#8902](https://github.com/tendermint/tendermint/pull/8902) Always start blocksync and avoid misconfiguration (@tychoish)
```
Comment (thanethomson) [inline CHANGELOG.md:15]: What does this mean?

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,35 @@
...
+
+### FEATURES
+
+- [cli] [\#8675] Add command to force compact goleveldb databases (@cmwaters)
```
Comment (thanethomson) [inline CHANGELOG.md:19]: Missing link.

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,35 @@
...
+
+### IMPROVEMENTS
+
+- [p2p] [\#8914](https://github.com/tendermint/tendermint/pull/8914) [\#8875](https://github.com/tendermint/tendermint/pull/8875) Improvements to peer dialing (backported). (@tychoish)
```
Comment (thanethomson) [inline CHANGELOG.md:23]: What improvements?

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,35 @@
...
+- [p2p] [\#8820](https://github.com/tendermint/tendermint/pull/8820) add eviction metrics and cleanup dialing error handling (backport #8819) (@tychoish)
```
Comment (thanethomson) [inline CHANGELOG.md:24]: Which metrics?

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,35 @@
...
+- [logging] [\#8896](https://github.com/tendermint/tendermint/pull/8896) Do not pre-process log results (backport #8895). (@tychoish)
```
Comment (thanethomson) [inline CHANGELOG.md:25]: What does this mean? Does this affect users in any way?

File: CHANGELOG.md [hunk 6]
```diff
@@ -2,6 +2,35 @@
...
+- [p2p] [\#8956](https://github.com/tendermint/tendermint/pull/8956) Simpler priority queue (backport #8929). (@tychoish)
```
Comment (thanethomson) [inline CHANGELOG.md:26]: What effect does this have on the P2P system?

File: CHANGELOG.md [hunk 7]
```diff
@@ -2,6 +2,35 @@
...
+
+### BUG FIXES
+
+- [mempool] [\#8944](https://github.com/tendermint/tendermint/pull/8944) Fix unbounded heap growth in the priority mempool. (@creachadair)
+- [p2p] [\#8869](https://github.com/tendermint/tendermint/pull/8869) Set empty timeouts to configed values. (backport #8847). (@williambanfield)
```
Comment (thanethomson) [inline CHANGELOG.md:31]: What is `configed`?

**Review by thanethomson — COMMENTED** [review]
What value does it add to have the `backport` signifiers in changelog entries? Is this something useful for the team, users, both?

File: CHANGELOG.md [hunk 8]
```diff
@@ -2,6 +2,35 @@
 
 Friendly reminder: We have a [bug bounty program](https://hackerone.com/cosmos).
 
+## v0.35.8
+
+July 12, 2022
+
+Special thanks to external contributors on this release: @joeabbey
```
Comment (thanethomson) [inline CHANGELOG.md:9]: Even if we don't elaborate further on specific changelog entries, can we please look at, from now on, adding a release summary for each release? It should focus on the main value-adds users would potentially be interested in. Something like:

```suggestion
This release primarily fixes a bug in the priority mempool that resulted in unbounded heap growth when under load.

As a temporary measure toward reducing disk space for a pruning node, this release also introduces a utility command to perform a compaction on GoLevelDB-based state and block stores. This utility command will be removed in future after the planned storage engine refactor.

Special thanks to external contributors on this release: @joeabbey
```

Comment (creachadair) [reply CHANGELOG.md:9]: That seems like a reasonable idea. Maybe consider making a PR to amend the RELEASES doc to include that.

File: CHANGELOG.md (hunk 5 above)
Comment (creachadair) [reply CHANGELOG.md:25]: It affects developers who use Tendermint's logging infrastructure (e.g., SDK applications). If you look at the original discussion, this was added to reduce memory pressure on hot logging paths.

File: CHANGELOG.md (hunk 7 above)
Comment (creachadair) [reply CHANGELOG.md:31]: The past participle of "to config" 😉

File: CHANGELOG.md (hunk 4 above)
Comment (creachadair) [reply CHANGELOG.md:24]: The details are in the linked pull request. If we want to discuss having more detailed release notes, we can certainly do that—but this is not the place to have that discussion.

File: CHANGELOG.md (hunk 3 above)
Comment (creachadair) [reply CHANGELOG.md:23]: The details are in the linked pull requests.

File: CHANGELOG.md (hunk 1 above)
Comment (creachadair) [reply CHANGELOG.md:15]: The details are in the linked pull request and its associated issue https://github.com/tendermint/tendermint/issues/8899.

File: CHANGELOG.md (hunk 6 above)
Comment (tychoish) [reply CHANGELOG.md:30]: It's not turned on by default in 0.35, so "provides another configuration option to support operational needs and avoid message starvation in bursty patterns."

**Review by tychoish — APPROVED** [review]

**Comment by williambanfield** [comment]
@creachadair can we add information about the `simple-priority` queue type? That is one of the more major improvements to the p2p issues that is present in this changeset. It's not on by default so we should call it out as an option for users experiencing a problem gossiping messages.

**Pull request merged**